@uportal/user-profile-menu/src/components/ProfileMenu.vue

```diff
@@ -0,0 +1,217 @@
+<template>
+  <div class="profile-menu-container" ref="userProfileMenu">
+    <button class="profile-trigger" @click="toggleMenu()">
+      <div
+        class="avatar"
+        :style="{
+          width: avatarSize,
+          height: avatarSize,
+          'font-size': avatarSize
+        }"
+      >
+        <img :src="avatarUrl" v-if="avatarUrl && !useInitials" />
+        <span class="fallback" v-else>{{ fallbackText }}</span>
+      </div>
+      <span>{{ displayText }}</span>
+    </button>
+    <div class="profile-dropdown" v-if="menuOpen">
+      <div class="profile-dropdown-section profile-dropdown-header">
+        <slot name="header"></slot>
+      </div>
+      <div class="profile-dropdown-section profile-dropdown-content">
+        <slot name="default"></slot>
+      </div>
+      <div class="profile-dropdown-section profile-dropdown-footer">
+        <slot name="footer"></slot>
+      </div>
+    </div>
+  </div>
+</template>
+<script>
+import oidc from '@uportal/open-id-connect';
+
+export default {
+  name: 'profile',
+  data() {
+    return {
+      menuOpen: false,
+      user: {
+        name: '',
+        family_name: '',
+        given_name: '',
+        image: null
+      },
+      dataLoaded: false,
+      hasError: false,
+      errorMessage: ''
+    };
+  },
+  computed: {
+    avatarUrl: function() {
+      return this.user.image;
+    },
+    displayText: function() {
+      return this.user.name;
+    },
+    fallbackText: function() {
+      return (
+        this.user.family_name.substr(0, 1) + this.user.given_name.substr(0, 1)
+      );
+    }
+  },
+  props: {
+    avatarSize: { type: String, default: '50px' },
```

Comment:
:question: what is the reasoning for making this an attribute vs a css variable?

Reply:
Good point! will fix